pkgs/servers/mail/public-inbox/default.nix

```diff
@@ -22,53 +77,86 @@ in
 
 buildPerlPackage rec {
   pname = "public-inbox";
-  version = "1.2.0";
+  version = "1.8.0";
 
   src = fetchurl {
-    url = "https://public-inbox.org/releases/public-inbox-${version}.tar.gz";
-    sha256 = "0sa2m4f2x7kfg3mi4im7maxqmqvawafma8f7g92nyfgybid77g6s";
+    url = "https://public-inbox.org/public-inbox.git/snapshot/public-inbox-${version}.tar.gz";
+    sha256 = "sha256-laJOOCk5NecIGWesv4D30cLGfijQHVkeo55eNqNKzew=";
   };
 
-  patches = [
-    (fetchpatch {
-      url = "https://public-inbox.org/meta/20200101032822.GA13063@dcvr/raw";
-      sha256 = "0ncxqqkvi5lwi8zaa7lk7l8mf8h278raxsvbvllh3z7jhfb48r3l";
-    })
-    ./0002-msgtime-drop-Date-Parse-for-RFC2822.patch
-  ];
-
   outputs = [ "out" "devdoc" "sa_config" ];
 
   postConfigure = ''
     substituteInPlace Makefile --replace 'TEST_FILES = t/*.t' \
         'TEST_FILES = $(shell find t -name *.t ${testConditions})'
+    substituteInPlace lib/PublicInbox/TestCommon.pm \
+      --replace /bin/cp ${coreutils}/bin/cp
   '';
 
   nativeBuildInputs = [ makeWrapper ];
 
   buildInputs = [
-    DBDSQLite EmailMIME IOSocketSSL IPCRun Plack PlackMiddlewareReverseProxy
-    SearchXapian TimeDate URI highlight
+    AnyURIEscape
+    DBDSQLite
+    DBI
+    EmailAddressXS
+    EmailMIME
+    highlight
+    IOSocketSSL
+    IPCRun
+    Inline
+    InlineC
+    ParseRecDescent
+    Plack
+    PlackMiddlewareReverseProxy
+    SearchXapian
+    TimeDate
+    URI
+    libgit2 # For Gcf2
+    man
   ];
 
-  checkInputs = [ git openssl xapian ];
+  doCheck = !stdenv.isDarwin;
+  checkInputs = [
+    MailIMAPClient
+    curl
+    git
+    openssl
+    pkg-config
+    sqlite
+    xapian
+  ] ++ lib.optionals stdenv.isLinux [
+    LinuxInotify2
+  ];
   preCheck = ''
     perl certs/create-certs.perl
+    export TEST_LEI_ERR_LOUD=1
+    export HOME="$NIX_BUILD_TOP"/home
+    mkdir -p "$HOME"/.cache/public-inbox/inline-c
   '';
 
   installTargets = [ "install" ];
   postInstall = ''
     for prog in $out/bin/*; do
-        wrapProgram $prog --prefix PATH : ${lib.makeBinPath [ git ]}
+        wrapProgram $prog --prefix PATH : ${lib.makeBinPath [
+          git
+          /* for InlineC */
+          gnumake
+          gcc
```

Comment:
Would it be a better option to hardcode these in InlineC?

Reply:
Hmm, [looks like](https://metacpan.org/pod/Inline::C) `InlineC` is not bound to any specific C compiler or `make` utility, so I'm not sure if that would be better. Likely a question to be addressed by maintainers of the Perl infrastructure. In the meantime the few other uses of `InlineC` in Nixpkgs seem to rely on the user to install those utilities system wide (eg. in `environment.systemPackages`).

`public-inbox` is run both as `systemd` services and on the command line (for administrative tasks), in both cases `git` must be available for sure, and I guess `InlineC` (and hence `gcc` and `gnumake`) are only used by the services, but I don't know `public-inbox`'s internals enough to be sure, and that can change in the future, that's why I'm adding them all in scope with that wrapper.

Reply:
> I'm not sure if that would be better. Likely a question to be addressed by maintainers of the Perl infrastructure.

They are pretty inactive.

We can't easily fix the compiler but probably make since gnumake is used almost everywhere but we can use the wrapper for now.